application/src/main/java/com/softserve/lv460/application/mapper/localDevice/LocalDeviceRequestMapper.java

```diff
@@ -0,0 +1,30 @@
+package com.softserve.lv460.application.mapper.localDevice;
+
+import com.softserve.lv460.application.dto.localDevice.LocalDeviceRequestDTO;
+import com.softserve.lv460.application.entity.LocalDevice;
+import com.softserve.lv460.application.mapper.Mapper;
+import com.softserve.lv460.application.service.LocationService;
+import lombok.AllArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Component;
+
+@AllArgsConstructor
+@Component
+public class LocalDeviceRequestMapper implements Mapper<LocalDevice, LocalDeviceRequestDTO> {
+    private ModelMapper modelMapper;
+    private LocationService locationService;
+
+    @Override
+    public LocalDevice toEntity(LocalDeviceRequestDTO dto) {
+        LocalDevice entity = modelMapper.map(dto, LocalDevice.class);
+        entity.setLocation(locationService.findOne(dto.getLocationId()));
```

Comment:
You can use modelMapper.configure() or modelMapper.addMappings() instead of setting Location manually, but it's your choice.